Add 'check' module

beckyjackson opened the pull request:
Resolves #70 

### `gizmos.check`

The `check` module validates a SQLite database for use with other `gizmos` modules. We recommend running your database through `gizmos.check` before using the other commands.
```
python3 -m gizmos.check [path-to-database]
```

This command will check that both the `prefix` and `statements` tables exist with valid columns as defined by [rdftab](https://github.com/ontodev/rdftab.rs). If those tables are valid, it checks the contents of `statements` to make sure that:
* all CURIEs use valid prefixes
* `stanza` is never a blank node
* `stanza`, `subject`, and `predicate` are never `NULL`

`check` will also warn on the following:
* missing index on `stanza` column
* full IRIs that use a base defined in `prefix`

All errors are logged, and if errors are found, the command will exit with status code `1`.

beckyjackson commented:
@jamesaoverton I've added the warning for the missing `stanza` index.

jamesaoverton commented:
I like it!

Two small problems: A stanza can be a blank node, in rare cases. A predicate can never be a blank node.

One more feature please: Starting around line 92, for a large table with problems we could get a lot of messages about specific rows. But I'd like `check` to return quickly even for large tables. So I'd like a limit on the number of messages about specific rows, with a default of maybe 10 (too low?), and an option to override.

Do you think it makes sense to use standard message table output in this case? Maybe that would be overkill?

beckyjackson commented:
I added a `--limit` option with a default of 10 and fixed the blank node issue.

I don't think this really requires standardized outputs, but if you want them I can add it in.

jamesaoverton reviewed (approved):
Great, thanks!
